hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -677,4 +714,27 @@ private void rollbackPendingCommits() {
     });
     return compactionInstantTimeOpt;
   }
+
+  /**
+   * Auto Clean service running concurrently.
+   */
+  private static class AutoCleanerService extends AbstractAsyncService {
+
+    private final HoodieWriteClient writeClient;
+    private final String cleanInstant;
+
+    private AutoCleanerService(HoodieWriteClient writeClient, String cleanInstant) {
+      this.writeClient = writeClient;
+      this.cleanInstant = cleanInstant;
+    }
+
+    @Override
+    protected Pair<CompletableFuture, ExecutorService> startService() {
+      ExecutorService executor = Executors.newFixedThreadPool(1);
```

Comment:
move to constructor and reuse for the duration of a HoodieWriteClient ?

Reply:
Move to constructor of AutoCleanerService